proposals/0000-ghc2021.rst

```diff
@@ -11,13 +11,35 @@ GHC 2021
 
 This this the PR accompanying the process to define the ``GHC2021`` language set, as outlined in https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0372-ghc-extensions.rst.
 
+Motivation
+----------
+
+The `GHC20xx process proposal <https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0372-ghc-extensions.rst#motivation>`_ motivates why we feel the need for ``GHC20xx`` extensions in general.
+
+``GHC2021`` is the first iteration of this process, which makes it a bit special: It is large, as it includes all the obvious, low-hanging fruit. And because it introduces the concept to our users, it heavily influences acceptance and adoption of the whole idea. Therefore, this iteration is on the slightly conservative side, excluding some extensions that almost made it.
+
+
 Proposed Change
 ---------------
 
-The ``GHC2021`` language extension set comprises of the following language extensions:
+The ``GHC2021`` language extension set comprises of the following language extensions, relative to `Haskell200`_:
+
+* Syntax: `EmptyCase`_, `PostfixOperators`_, `MonadFailDesugaring`_, `TupleSections`_, `ImportQualifiedPost`_,  `NamedFieldPuns`_, `BangPatterns`_
+* Literals: `BinaryLiterals`_, `HexFloatLiterals`_, `NumericUnderscores`_
+* Types:  `GADTSyntax`_, `RankNTypes`_, `StarIsType`_, `TypeApplications`_, `PolyKinds`_, `StandaloneKindSignatures`_, `ExistentialQuantification`_, `TypeOperators`_, `ConstraintKinds`_
+* Type signatures: `ExplicitForAll`_, `KindSignatures`_, `NamedWildCards`_, `ScopedTypeVariables`_
+* Class and instances declarations: `FlexibleContexts`_, `FlexibleInstances`_, `MultiParamTypeClasses`_, `ConstrainedClassMethods`_, `InstanceSigs`_, `TypeSynonymInstances`_
+* Deriving mechanism: `DeriveDataTypeable`_, `DeriveFoldable`_, `DeriveFunctor`_, `DeriveTraversable`_, `StandaloneDeriving`_, `EmptyDataDeriving`_, `DeriveLift`_, `GeneralisedNewtypeDeriving`_, `DeriveGeneric`_
 
-* Everything in ``Haskell2010``, plus
-* TBD
+As a reminder, the following extensions are part of `Haskell2010`_, and continue to be enabled:
+
+* ``ImplicitPrelude``, ``StarIsType``, ``MonomorphismRestriction``, ``TraditionalRecordSyntax``, `EmptyDataDecls`_, `ForeignFunctionInterface`_, ``PatternGuards``, ``DoIfThenElse``,  ``RelaxedPolyRec``.
```

Comment:
GHC's interpretation of Haskell2010 includes `MonadFailDesugaring`.

(Also `StarIsType` is in both lists: in to be included and this remainder).

Reply:
> GHC's interpretation of Haskell2010 includes `MonadFailDesugaring`.

Although not listed under https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/control.html?highlight=haskell2010#extension-Haskell2010. Even if it isn’t an technical change of status quo, I think it makes sense to treat it like a change to the specification. It’s all a bit squshiy when it comes to report vs. `base` issues…

Thanks for `StarIsType`, fixed.

Reply:
It won't. It's then better to omit whole extension, as it will be removed from future GHCs. See https://gitlab.haskell.org/ghc/ghc/-/merge_requests/4767

Reply:
If we _don’t_ list it in this section it looks as thought the committee voted against `MonadFailDesugaring`. I think this is fine, if at some point _flag_ goes away in GHC, effectively turning it on constantly, that’s in accordance with what GHC2021 says it should look like, doesn't it?

Reply:
I argue that the flag is turned on for the GHC interpretation of Haskell2010, which is what I understand `Haskell2010` refers to. I think we should be very clear that GHC Haskell 2010 diverges from report *considerably*, and there is no point trying to hide that. 

https://ghc.gitlab.haskell.org/ghc/doc/users_guide/bugs.html#divergence-from-haskell-98-and-haskell-2010 is lengthy.

Reply:
I clarified in the proposal text; better now?